src/components/product/ProductDetailCard.js

```diff
@@ -1,31 +1,52 @@
 import React from 'react';
-import { Card, Button } from '../UI/';
+import { Card, Button, ErrorSpan } from '../UI/';
 import PropTypes from 'prop-types';
+import { useProductDetail } from '../../utils/hooks/useProductDetail';
 
 function ProductDetailCard({ product }) {
+  const {
+    stock,
+    amount,
+    handleChange,
+    error,
+    addToCart,
+    disabled
+  } = useProductDetail(product);
+
   return (
     <Card padding="2rem" cursor="auto">
       <h2>{product.data?.name}</h2>
 
-      <p className="price">$ {product.data?.price}</p>
-      <span className="category">{product.data?.category?.slug}</span>
-      <p>SKU: {product?.data?.sku}</p>
+      <p className="price">$ {product.data.price}</p>
+      <span className="category">{product.data.category.slug}</span>
+      <p>SKU: {product.data.sku}</p>
```

Comment:
veo que usas la información del producto en varias partes, podemos deconstruir la info para omitir el `product.`

Si agregamos algo como:

```jsx
const { data: { price, category: { slug }, sku } } = product;
```

podemos cambiar esta parte:

```suggestion
      <p className="price">$ {price}</p>
      <span className="category">{slug}</span>
      <p>SKU: {sku}</p>
```